python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -8315,3 +8315,57 @@ def test_dataframe_roundtrip_arrow_struct_dtype(gdf):
     expected = gd.DataFrame.from_arrow(table)
 
     assert_eq(gdf, expected)
+
+
+@pytest.mark.parametrize(
+    "data", [{"a": [1, 2, 3], "b": [4, 5, 6], "c": [7, 8, 9]},]
+)
+@pytest.mark.parametrize(
+    "index", [{0: 1, 1: 2, 2: 3}],
+)
+@pytest.mark.parametrize(
+    "level", ["x", 0],
+)
+def test_dataframe_MI_for_level(data, index, level):
+    pdf = pd.DataFrame(
+        data,
+        index=pd.MultiIndex.from_tuples([(0, 1, 2), (1, 2, 3), (2, 3, 4)]),
+    )
+    pdf.index.names = ["x", "y", "z"]
+    gdf = gd.from_pandas(pdf)
+
+    expect = pdf.rename(index=index, level=level)
+    got = gdf.rename(index=index, level=level)
+
+    assert_eq(expect, got)
+
+
+@pytest.mark.parametrize(
+    "data", [{"a": [1, 2, 3], "b": [4, 5, 6], "c": [7, 8, 9]},]
+)
+@pytest.mark.parametrize(
+    "columns", [{"a": "f", "b":"g"}],
+)
+@pytest.mark.parametrize(
+    "level", [0,1],
+)
+def test_dataframe_with_MC_for_level(data, columns, level):
+    gdf = gd.DataFrame(data)
+    gdf.columns = pd.MultiIndex.from_tuples([("a", 1), ("a", 2), ("b", 1)])
+
+    pdf = gdf.to_pandas()
+
+    expect = pdf.rename(columns=columns, level=level)
+    got = gdf.rename(columns=columns, level=level)
+
+    assert_eq(expect, got)
+
+
+def test_dataframe_with_RI_for_level():
```

Comment:
but for this case, i am testing level support for regular (non-multicolumns) so i don't understand why i should test rename on index parameter?  please clarify

Reply:
I take that as a corner case where dataframe with regular index should also support specifying `level=0`, such as
```
>>> pdf = pd.DataFrame({'a': [1,2,3], 'b':[4,5,6], 'c':[7,8,9]})
>>> pdf.rename(index={1:123}, level=0)
     a  b  c
0    1  4  7
123  2  5  8
2    3  6  9
```
Especially in this case, I'm interested because the index is a `RangeIndex`, renaming it would require instantiating it into a `Int64Index`.